ratis-server/src/main/java/org/apache/ratis/server/leader/LogAppenderBase.java

```diff
@@ -53,6 +57,8 @@ public abstract class LogAppenderBase implements LogAppender {
   private final LogAppenderDaemon daemon;
   private final AwaitForSignal eventAwaitForSignal;
 
+  private final BlockingQueue<Consumer<AppendEntriesReplyProto>> watcherList;
```

Comment:
Why using a BlockingQueue?  It seems that a List is good enough.

Reply:
Done, I'll use ArrayList instead.